src/Common/BSONCXXHelper.h

```diff
@@ -23,13 +27,43 @@ using bsoncxx::builder::basic::document;
 using bsoncxx::builder::basic::kvp;
 using bsoncxx::builder::basic::make_document;
 
-static bsoncxx::types::bson_value::value fieldAsBSONValue(const Field & field, const DataTypePtr & type)
+static bsoncxx::types::bson_value::value tryOid(const std::string & str)
+{
+    try
+    {
+        return bsoncxx::oid(str);
+    } catch (bsoncxx::v_noabi::exception & e)
```

Comment:
(For some reason, my clangd doesn't index the bsoncxx headers, so I can't check by myself.)

So l. 32-40 misuse exceptions as part of the control flow which is an anti-pattern. Is there maybe an alternative function in bsoncxx which returns a return code instead of an exception?